File: frontend/src/api/interface/operation-log.ts
import { DateTimeFormats } from '@intlify/core-base';

export interface ResOperationLog {
    id: number;
    group: string;
    source: string;
    action: string;
    ip: string;
    path: string;
    method: string;
    userAgent: string;
    body: string;
    resp: string;

    status: number;
    latency: number;
    errorMessage: string;

    detail: string;
    createdAt: DateTimeFormats;
}

File: backend/app/dto/operation_log.go
package dto

import (
	"time"
)

type OperationLogBack struct {
	ID     uint   `json:"id"`
	Group  string `json:"group"`
	Source string `json:"source"`
	Action string `json:"action"`

	IP        string `json:"ip"`
	Path      string `json:"path"`
	Method    string `json:"method"`
	UserAgent string `json:"userAgent"`
	Body      string `json:"body"`
	Resp      string `json:"resp"`

	Status       int           `json:"status"`
	Latency      time.Duration `json:"latency"`
	ErrorMessage string        `json:"errorMessage"`

	Detail    string    `json:"detail"`
	CreatedAt time.Time `json:"createdAt"`
}
